### PR TITLE
Profiling tool can miss datasources when they are GPU reads

### PR DESCRIPTION
**Pull request by tgravescs**
fixes https://github.com/NVIDIA/spark-rapids/issues/4759

This fixes it so we properly report GPU based datasources (csv, parquet, json, orc) when the profiling tool looks at the event logs from a run witth rapids plugin enabled. Tested with both dsv1 and dsv2 versions.  This also changes JDBC to report the other fields like format, locations, etc.

It also fixes a bug with CSV files where there could be commas in the field even though our delimiter is a comma. That makes it so if you read the CSV file back into Spark it truncates that. Specifically this happens with the file schema where format is: name:Type,name2:Type2,...   So for this we took the same logic used by the qualiciation tool to just replace the comma in any strings.

example output:

```
Data Source Information:
+--------+-----+---------------+--------------------------------------------------------------------------------------------------------------------+-------------+---------------------------------------------------------------------------------------------+
|appIndex|sqlID|format         |location                                                                                                            |pushedFilters|schema                                                                                       |
+--------+-----+---------------+--------------------------------------------------------------------------------------------------------------------+-------------+---------------------------------------------------------------------------------------------+
|1       |0    |Text           |InMemoryFileIndex[file:/home/tgraves/workspace/spark-rapids-another/integration_tests/src/test/resources/people.csv]|[]           |value:string                                                                                 |
|1       |1    |gpucsv(GPU)    |InMemoryFileIndex[file:/home/tgraves/workspace/spark-rapids-another/integration_tests/src/test/re...                |unknown      |_c0:string,_c1:string,_c2:string                                                             |
|1       |2    |gpujson(GPU)   |InMemoryFileIndex[file:/home/tgraves/workspace/spark-rapids-another/integration_tests/target/test...                |unknown      |number:double                                                                                |
|1       |3    |gpuparquet(GPU)|InMemoryFileIndex[file:/home/tgraves/workspace/spark-rapids-another/integration_tests/target/test...                |[]           |loan_id:bigint,orig_channel:string,seller_name:string,orig_interest_rate:double,orig_upb:i...|
|1       |4    |gpuorc(GPU)    |InMemoryFileIndex[file:/home/tgraves/workspace/spark-rapids-another/tests/src/test/resources/file...                |[]           |loan_id:bigint,orig_channel:int,orig_interest_rate:double,orig_upb:int,orig_loan_term:int,...|
|2       |0    |Text           |InMemoryFileIndex[file:/home/user1/workspace/spark-rapids-another/integration_tests/src/test/resources/people.csv]  |[]           |value:string                                                                                 |
|2       |1    |CSV(GPU)       |InMemoryFileIndex[file:/home/user1/workspace/spark-rapids-another/integration_tests/src/test/re...                  |[]           |_c0:string,_c1:string,_c2:string                                                             |
|2       |2    |JSON(GPU)      |InMemoryFileIndex[file:/home/user1/workspace/spark-rapids-another/integration_tests/target/test...                  |[]           |number:double                                                                                |
|2       |3    |ORC(GPU)       |InMemoryFileIndex[file:/home/user1/workspace/spark-rapids-another/tests/src/test/resources/file...                  |[]           |loan_id:bigint,orig_channel:int,orig_interest_rate:double,orig_upb:int,orig_loan_term:int,...|
|2       |4    |Parquet(GPU)   |InMemoryFileIndex[file:/home/user1/workspace/spark-rapids-another/integration_tests/target/test...                  |[]           |loan_id:bigint,monthly_reporting_period:string,servicer:string,interest_rate:double,curren...|
+--------+-----+---------------+--------------------------------------------------------------------------------------------------------------------+-------------+---------------------------------------------------------------------------------------------+

```

### CODE REVIEW

**Comment by tgravescs** [comment]
build

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala [hunk 1]
```diff
@@ -176,62 +167,62 @@ abstract class AppBase(
...
+    if (endIndex != -1) {
+      subStr.substring(0, endIndex)
+    } else {
+      subStr
     }
   }
 
   // This will find scans for DataSource V2, if the schema is very large it
   // will likely be incomplete and have ... at the end.
   protected def checkGraphNodeForBatchScan(sqlID: Long, node: SparkPlanGraphNode): Unit = {
```
Comment (nartal1) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala:186]: This method now checks more than just BatchScan. I am not sure if the method name needs to be changed.

**Comment by nartal1** [comment]
Just a question on method name. Rest all LGTM.

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala (hunk 1 above)
Comment (tgravescs) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala:186]: I don't think its a big deal but I'll update to Reads

**Comment by tgravescs** [comment]
build

**Review by nartal1 — APPROVED** [review]

**Pull request merged**